Makefile

```
@@ -40,10 +42,32 @@ docker-push:
 	@echo "Pushing Docker image to Docker Hub..."
 	docker push $(DOCKER_IMAGE):latest
 
-# Deploy target (build and push Docker image)
+# Deploy target (build and push Docker image to Docker Hub)
 .PHONY: deploy
 deploy: docker-build docker-push
 
+# Create and run a local Docker registry
+.PHONY: local-registry
+local-registry:
```

nit: namespace this to `docker`. As in:
```suggestion
docker-registry-local:
```

That way, all docker commands start with `docker`. Also not entirely sure if the `||` is correct. It could fail for ANY reason (docker isn't running, image can't be pulled (no internet), container with that name is already running).

It is better to have no unhappy-path output than incorrect unhappy-path output.